src/lib/InstantSearch.js

```diff
@@ -339,7 +339,10 @@ class InstantSearch extends EventEmitter {
    */
   dispose() {
     this.removeWidgets(this.widgets);
+    // allow to restart this instance
     this.started = false;
+    // start from a fresh state, not previously set refinements
```

Comment:
I had something like this in mind.

```suggestion

    // The helper needs to be reset to perform the next search from a fresh state.
    // If not reset, it would use the state stored before calling `dispose()`.
```

Documenting this way is a pattern that I would like to use more in the codebase. It makes the intent clear and explains why a line is required. WDYT?

Reply:
As you want, I didn't realise you preferred it written in full